SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -627,35 +710,209 @@ void ASpatialDebugger::DrawDebug(UCanvas* Canvas, APlayerController* /* Controll
 
 			if (Actor != nullptr)
 			{
-				FVector ActorLocation = Actor->GetActorLocation();
-
-				if (ActorLocation.IsZero())
+				FVector2D ScreenLocation = ProjectActorToScreen(Actor, PlayerLocation);
+				if (ScreenLocation.IsZero())
 				{
 					continue;
 				}
 
-				if (FVector::Dist(PlayerLocation, ActorLocation) > MaxRange)
+				DrawTag(Canvas, ScreenLocation, EntityId, Actor->GetName(), true /*bCentre*/);
+			}
+		}
+	}
+}
+
+void ASpatialDebugger::SelectActorsToTag(UCanvas* Canvas)
+{
+	if (LocalPlayerController.IsValid())
+	{
+		FVector2D NewMousePosition;
+
+		if (LocalPlayerController->GetMousePosition(NewMousePosition.X, NewMousePosition.Y))
+		{
+			if (CrosshairTexture != nullptr)
+			{
+				// Display a crosshair icon for the mouse cursor
+				// Offset by half of the texture's dimensions so that the center of the texture aligns with the center of the Canvas.
+				FVector2D CrossHairDrawPosition(NewMousePosition.X - (CrosshairTexture->GetSurfaceWidth() * 0.5f),
+												NewMousePosition.Y - (CrosshairTexture->GetSurfaceHeight() * 0.5f));
+
+				// Draw the crosshair at the mouse position.
+				FCanvasTileItem TileItem(CrossHairDrawPosition, CrosshairTexture->Resource, FLinearColor::White);
+				TileItem.BlendMode = SE_BLEND_Translucent;
+				Canvas->DrawItem(TileItem);
+			}
+
+			TWeakObjectPtr<AActor> NewHoverActor = GetActorAtPosition(NewMousePosition);
+
+			if (NewHoverActor == nullptr)
+			{
+				// No actor under the cursor so revert hover materials on previous actor
+				RevertHoverMaterials();
+			}
+			else if (NewHoverActor != HoverActor)
+			{
+				// New actor under the cursor
+
+				// Revert hover materials on previous actor
+				RevertHoverMaterials();
+
+				// Set hover materials on new actor
+				ActorMeshComponents = NewHoverActor->GetComponentsByClass(UMeshComponent::StaticClass());
+				for (UActorComponent* NewActorMeshComponent : ActorMeshComponents)
 				{
-					continue;
+					UMeshComponent* NewActorStaticMeshComponent = Cast<UMeshComponent>(NewActorMeshComponent);
+					// Store previous materials
+					ActorMeshMaterials.Add(NewActorStaticMeshComponent->GetMaterial(0));
+					// Set wireframe material on new actor
+					NewActorStaticMeshComponent->SetMaterial(0, WireFrameMaterial);
 				}
+				HoverActor = NewHoverActor;
+			}
 
-				FVector2D ScreenLocation = FVector2D::ZeroVector;
-				if (LocalPlayerController.IsValid())
+			// Draw tags above selected actors
+			for (TWeakObjectPtr<AActor> SelectedActor : SelectedActors)
+			{
+				if (SelectedActor.IsValid())
 				{
-					SCOPE_CYCLE_COUNTER(STAT_Projection);
-					UGameplayStatics::ProjectWorldToScreen(LocalPlayerController.Get(), ActorLocation + WorldSpaceActorTagOffset,
-														   ScreenLocation, false);
+					if (const Worker_EntityId_Key* HitEntityId = EntityActorMapping.FindKey(SelectedActor))
+					{
+						FVector PlayerLocation = GetLocalPawnLocation();
+
+						if (SelectedActor != nullptr)
```

Comment:
Do this check with the IsValid check?

Reply:
Afaik IsValid means it isn't null, so this check shouldn't be necessary.

Reply:
Done an IsValid check earlier so removed the nullptr check as it is redundant.